File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {

        for (int i = digits.length - 1; i >= 0; i--) {
            digits[i]++;

            if (digits[i] < 10)
                return digits;
            else
                digits[i] = 0;
        }

        if(digits[0]==0){
            int[] result = new int[digits.length+1];
            result[0]=1;
            
            for(int i = 1; i<result.length; i++)
                result[i] = digits[i-1];
            
            return result;
        }

        return digits;
    }
}

File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        plus = 1
        length = len(digits)

        for i in range(len(digits)):
            digits[length-i-1] += plus

            plus = digits[length-i-1]//10
            digits[length-i-1] %= 10

        if plus != 0:
            digits = [plus]+digits
        
        return digits